src/main/java/io/nats/client/support/JsonParser.java

```diff
@@ -0,0 +1,363 @@
+                    }
+                    break;
+                case '}':
+                    return map;
+                default:
+                    throw new JsonParseException("Expected a ',' or '}'.");
+            }
+        }
+    }
+
+    private char nextToken() {
+        peekToken();
+        idx = nextIdx;
+        nextIdx = -1;
+        previous = current;
+        current = next;
+        next = 0;
+        return current;
+    }
+
+    private char nextChar() {
+        previous = current;
+        if (idx == len) {
+            current = 0;
+        }
+        else {
+            current = json.charAt(idx++);
+        }
+        next = 0;
+        nextIdx = -1;
+        return current;
+    }
+
+    private char peekToken() {
+        if (nextIdx == -1) {
+            nextIdx = idx;
+            next = 0;
+            while (nextIdx < len) {
+                char c = json.charAt(nextIdx++);
+                switch (c) {
+                    case ' ':
+                    case '\r':
+                    case '\n':
+                    case '\t':
+                        continue;
+                }
+                return next = c;
+            }
+        }
+        return next;
+    }
+
+    // next string assumes you have already seen the starting quote
+    private String nextString() throws JsonParseException {
+        StringBuilder sb = new StringBuilder();
+        while (true) {
+            char c = nextChar();
+            switch (c) {
+                case 0:
+                case '\n':
+                case '\r':
+                    throw new JsonParseException("Unterminated string.");
+                case '\\':
+                    c = nextChar();
+                    switch (c) {
+                        case 'b':
+                            sb.append('\b');
+                            break;
+                        case 't':
+                            sb.append('\t');
+                            break;
+                        case 'n':
+                            sb.append('\n');
+                            break;
+                        case 'f':
+                            sb.append('\f');
+                            break;
+                        case 'r':
+                            sb.append('\r');
+                            break;
+                        case 'u':
+                            sb.append(parseU());
+                            break;
+                        case '"':
+                        case '\'':
+                        case '\\':
+                        case '/':
+                            sb.append(c);
+                            break;
+                        default:
+                            throw new JsonParseException("Illegal escape.");
+                    }
+                    break;
+                default:
+                    if (c == '"') {
+                        if (sb.toString().equals("subjects")) {
```

Comment:
This block seems like it could be removed?

Reply:
yeah, that was debugging.